zerver/decorator.py

```diff
@@ -110,7 +104,10 @@ def wrapper(request: HttpRequest, *args: object, **kwargs: object) -> HttpRespon
             err_method = request.method
             logging.warning('Method Not Allowed (%s): %s', err_method, request.path,
                             extra={'status_code': 405, 'request': request})
-            return HttpResponseNotAllowed(["POST"])
+            if request.error_format == 'JSON':
+                return json_method_not_allowed(["POST"])
+            else:  # nocoverage
+                return TemplateResponse(request, "404.html", context={'status_code': 405})
```

Comment:
You should add a test for this, perhaps in `test_urls.py`, rather than adding a `nocoverage` comment.

otherwseiotherwise, lgtm!